sklearn/multioutput.py

```diff
@@ -0,0 +1,214 @@
+            ovr = OneVsRestClassifier(estimator, self.n_jobs)
+            self.estimators_[i] = ovr.fit(X, y[:, i])
+
+        return self
+
+    def predict(self, X):
+        """Predict multi-class multiple target variable using a model
+         trained for each target variable.
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+
+        Returns
+        -------
+        y : (sparse) array-like, shape = [n_samples, p_targets]
+            Multi-output targets. An indicator matrix turns on multilabel
+            classification.
+            Predicted multi-class targets across multiple predictors.
+            Note:  entirely separate models are generated
+            for each predictor.
+        """
+        # check to see if the fit has been performed
+        check_is_fitted(self, 'estimators_')
+
+        results =[]
+        for i in range(self._num_outputs):
+            estimator = self.estimators_[i]
+            results.append(estimator.predict(X))
+        return np.asarray(results)
+
+    def predict_proba(self, X):
+        """Probability estimates. This returns prediction probabilites
+        for each class for each label in the form of a dictionary.      
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+
+        Returns
+        -------
+        T : (sparse) array-like, shape = [n_samples, p_targets]
+            Returns the probability of the sample for each target in the model.
+            
+        """
+        # check to see whether the fit has occured.
+        check_is_fitted(self, 'estimators_')
+
+        results =[]
+        for label, model_ in self.estimators_.iteritems():
+            results.append(model_.predict_proba(X))
+        return np.asarray(results)
+
+    def score(self, X, Y):
+        """"Returns the mean accuracy on the given test data and labels.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+        Y : array-like, shape = [n_samples, p_targets]
+
+        Returns
+        -------
+        scores : array-like, shape = [n_samples, p_targets]
+            Array of p_target floats of the mean accuracy
+            of each estimator_.predict wrt. y.
+        """
+        check_is_fitted(self, 'estimators_')
+
+        num_outputs = Y.shape[1]
+
+        # Score the results for each function
+        results =[]
+        for i in range(num_outputs):
+            estimator = self.estimators_[i]
+            results.append(estimator.score(X,Y[:,i]))
+        return np.asarray(results)
+
+    def get_params(self, deep=True):
+        return {"estimator": self.estimator, "n_jobs": self.n_jobs}
+
+    def set_params(self, **parameters):
+        for parameter, value in six.iteritems(parameters):
+            self.setattr(parameter, value)
+        return self
+
+    def __repr__(self):
```

Comment:
you need not do this. inheriting from `BaseEstimator` automatically does it for you